src/librustc_borrowck/diagnostics.rs

```diff
@@ -314,6 +314,59 @@ let c = &i; // still ok!
 ```
 "##,
 
+E0504: r##"
+This error occurs when an attempt is made to move a borrowed variable into a
+closure. For example:
+
+```compile_fail
```

Comment:
What do you think about changing the `Rc` example into an `Arc` example and using `thread::spawn` there? It would highlight threads as well as giving a more motivating example for using reference counting (compared with the existing example, which uses reference counting for no reason).